torch/quantization/fx/quantization_patterns.py

```diff
@@ -82,11 +83,20 @@ def _maybe_get_last_node_only_observer(
         If the last node of the pattern is observed, return the observer
         instance. Otherwise, return None.
         """
-        for maybe_obs_node, _ in self.last_node.users.items():
+        last_node = self.last_node
+
+        # Handle the additional mul node that gets added during equalization if
```

Comment:
Why is this a special case here?

Reply:
The reason why there is a special casing when there is a quantized but not equalized node followed by an equalized node. 

For example, let's say linear1 is not being equalized but linear2 is being equalized. The following graph is what we get after prepare:
```
x -> inp_quant_obs1 -> linear1 -> out_quant_obs1 -> inp_eq_obs2 -> linear2 -> out_quant_obs2
```

After the equalization functions, we have:
```
x -> inp_quant_obs1 -> linear1 -> mul -> out_quant_obs1 (scaled) -> linear2 -> out_quant_obs2
```

In this current function (`maybe_get_last_node_observer`), it originally only looped through the op node's users to find its output quantization observer. However after applying the equalization nodes, linear1's output quantization observer no longer appears directly after linear1, so I added some extra casing.

It should not interfere with the case where the model already has an op node, because in that case I think there would be a output quantization observer inserted before the mul node:
```
x -> inp_quant_obs1 -> linear1 -> out_quant_obs1 -> mul -> out_quant_obs2
```

Reply:
What about the case where linear1 isn't being quantized and we have a mul node after it. Would we end up calling this function in that case?

@jerryzh168 could you help take a look at this change and see if it is okay for the non-equalization use case?

Reply:
right, there is a chance that the mul following linear is not quantized, for example user explicitly configures that to be not quantized through qconfig_dict, I feel this is probably a bit too hacky, let's think about if we have a different workaround

Reply:
> What about the case where linear1 isn't being quantized and we have a mul node after it. Would we end up calling this function in that case?

If linear1 isn't being quantized, then I think it wouldn't enter this function.

> the mul following linear is not quantized, for example user explicitly configures that to be not quantized through qconfig_dict.

In this case, I think the structure of the prepared graph would still be, so the mul would not appear between the linear and output quantization observer if I'm not mistaken:
```
x -> inp_quant_obs1 -> inp_eq_obs -> linear1 -> out_quant_obs1 -> mul
```

Reply:
what if linear is also not quantized and the op following mul is quantized? would we have 
```
linear1 -> mul -> observer -> conv -> ...
```   and when we call `_maybe_get_last_node_only_observer` it would return the observer?

Reply:
I see, I understand that case now. Could we check to see if the mul operator contains a node with the name "equalization_scale" as an argument? Or I could modify the name of the mul node for equalization to be like "mul_for_equalization" and check if the mul operator found contains "equalization" in the name?

Reply:
> Could we check to see if the mul operator contains a node with the name "equalization_scale" as an argument?

This seems okay to me. Can we add an argument to this function to denote equalization is enabled, in which case it checks for this argument for a `mul`?